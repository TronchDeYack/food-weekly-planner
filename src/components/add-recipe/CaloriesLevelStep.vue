<template>
  <div>
    <StepTitle :complete="complete" :step="step" :title="title" />
    <StepContent :step="step" @next-step="nextStep" @previous-step="previousStep">
      <v-slider
        :tick-labels="caloriesLabel"
        :max="caloriesLabel.length - 1"
        step="1"
        ticks="always"
        tick-size="3"
      ></v-slider>
    </StepContent>
  </div>
</template>

<script>
import StepTitle from '@/components/add-recipe/StepTitle.vue';
import StepContent from '@/components/add-recipe/StepContent.vue';

export default {
  name: 'add-recipe-calories-level-step',
  components: {
    StepContent,
    StepTitle,
  },
  props: {
    title: {
      type: String,
    },
    step: {
      type: Number,
    },
    complete: {
      type: Boolean,
    },
  },
  data() {
    return {
      caloriesLabel: [
        'Low',
        'Medium',
        'High',
      ],
    };
  },
  // TODO: remove that by using a store
  methods: {
    nextStep(event) {
      this.$emit('next-step', event);
    },
    previousStep(event) {
      this.$emit('previous-step', event);
    },
  },
};
</script>
