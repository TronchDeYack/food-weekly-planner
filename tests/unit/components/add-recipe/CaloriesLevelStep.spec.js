import CaloriesLevelStep from '@/components/add-recipe/CaloriesLevelStep.vue';

import { createComponent } from '../../../utils';

describe('CaloriesLevelStep', () => {
  const propsData = {
    title: 'EAT ME',
    step: 2,
    complete: false,
  };

  let wrapper;

  beforeEach(() => {
    wrapper = createComponent(CaloriesLevelStep, propsData);
  });

  describe('render', () => {
    it('Should render.', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('state', () => {
    it('Should initialize "caloriesLabel" to an array.', () => {
      expect(wrapper.vm.$data.caloriesLabel).toEqual(['Low', 'Medium', 'High']);
    });
  });
});
