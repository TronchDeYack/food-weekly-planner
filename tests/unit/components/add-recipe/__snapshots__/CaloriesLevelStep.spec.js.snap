// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaloriesLevelStep render Should render. 1`] = `
<div>
  <steptitle-stub title="EAT ME" step="2"></steptitle-stub>
  <stepcontent-stub step="2">
    <v-slider-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" loaderheight="2" max="2" min="0" step="1" thumbsize="32" ticklabels="Low,Medium,High" ticks="always" ticksize="3"></v-slider-stub>
  </stepcontent-stub>
</div>
`;
